src/plugins/data/common/es_query/kuery/node_types/wildcard.ts

```diff
@@ -18,22 +18,23 @@
  */
 
 import { fromLiteralExpression } from '../ast/ast';
+import { WildcardTypeBuildNode } from './types';
 
 export const wildcardSymbol = '@kuery-wildcard@';
 
 // Copied from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
-function escapeRegExp(string) {
-  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+function escapeRegExp(str: string) {
+  return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
 }
 
 // See https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_reserved_characters
-function escapeQueryString(string) {
-  return string.replace(/[+-=&|><!(){}[\]^"~*?:\\/]/g, '\\$&'); // $& means the whole matched string
+function escapeQueryString(str: string) {
+  return str.replace(/[+-=&|><!(){}[\]^"~*?:\\/]/g, '\\$&'); // $& means the whole matched string
 }
 
-export function buildNode(value) {
+export function buildNode(value: string): WildcardTypeBuildNode {
   if (!value.includes(wildcardSymbol)) {
-    return fromLiteralExpression(value);
+    return (fromLiteralExpression(value) as unknown) as WildcardTypeBuildNode;
```

Comment:
Instead of doing `as unknown) as WildcardTypeBuildNode` casting, would it work to have this function's return value be `WildcardTypeBuildNode | KueryNode`, since `KueryNode` is the return type of `fromLiteralExpression`?